DOM Virtual e Objetos Internos (#159)
reactjs/pt-BR.reactjs.org: content/docs/faq-internals.md

---
id: faq-internals
title: Virtual DOM e Objetos Internos
permalink: docs/faq-internals.html
layout: docs
category: FAQ
---

> ### O que é o Virtual DOM?

Reviewer: Faltou a referência para a ancora:
```suggestion
### O que é o Virtual DOM? {#what-is-the-virtual-dom}
```

Author: nao tinha visto tambem kkkk

> ### O que é um Virtual DOM?

Reviewer 2: ```suggestion
### O que é o Virtual DOM?
```

`the` é artigo definido

O DOM Virtual (VDOM) é um conceito de programação onde uma representação ideal, ou "virtual", da interface do usuário é mantida em memória e sincronizada com o DOM "real" por uma biblioteca como o ReactDOM. Esse processo é chamado de [reconciliação](/docs/reconciliation.html).

> Essa aproximação permite a API declarativa do React: Você diz ao React qual o state que você quer que a interface do usuário esteja, e ele garante que o DOM seja igual ao state. Isso abstrai a manipulação de atributos, handling de eventos e atualização manual do DOM que, caso ao contrario, você teria que usar para construir o seu app.

Reviewer: `handling de eventos` ficou um pouco  confuso traduzido

Author: arrumei tudo como virtual e handling para manipulação

> Essa aproximação permite a API declarativa do React: Você diz ao React qual o state que você quer que a interface do usuário esteja, e ele garante que o DOM seja igual ao state. Isso abstrai a manipulação de atributos, manipulação de eventos e atualização manual do DOM que, caso ao contrario, você teria que usar para construir o seu app.

Reviewer 2: ```suggestion
Essa abordagem permite a API declarativa do React: Você diz ao React qual o state que você quer que a interface do usuário esteja, e ele garante que o DOM seja igual ao state. Isso abstrai a manipulação de atributos, manipulação de eventos e atualização manual do DOM que, caso ao contrario, você teria que usar para construir o seu app.
```

> Desde que o "virtual DOM" é mais um padrão do que uma tecnologia específica, as pessoas às vezes o citam para significar coisas diferentes. No mundo do React, o termo "virtual DOM" é geralmente associado aos [Elementos do React](/docs/rendering-elements.html) uma vez que eles são objetos representando a interface do usuário. O React, contudo, também usa objetos internos chamados "fibers" para segurar informações adicionais sobre a árvore de componentes. Eles também podem ser considerados parte da implementação do "virtual DOM" no React.

Reviewer 2: ```suggestion
Dado que "virtual DOM" é mais um padrão do que uma tecnologia específica, as pessoas às vezes o citam querendo dizer coisas diferentes. No mundo do React, o termo "virtual DOM" é geralmente associado aos [Elementos do React](/docs/rendering-elements.html) uma vez que eles são objetos representando a interface do usuário. O React, contudo, também usa objetos internos chamados "fibers" para conter informações adicionais sobre a árvore de componentes. Eles também podem ser considerados parte da implementação do "virtual DOM" no React.
```

### O Shadow DOM é a mesma coisa que o Virtual DOM? {#is-the-shadow-dom-the-same-as-the-virtual-dom}

> Não, eles são diferentes. O Shadow DOM é uma tecnológica do navegador desenhada principalmente para escopar variáveis e CSS em componentes web. O virtual DOM é um conceito implementado por bibliotecas em JavaScript em cima das APIs do navegador.

Reviewer 2: ```suggestion
Não, eles são diferentes. O Shadow DOM é uma tecnologia do navegador desenhada principalmente para conter variáveis e CSS no escopo de web components. O virtual DOM é um conceito implementado por bibliotecas em JavaScript em cima das APIs do navegador.
```

Reviewer 2: Optei por manter o "web components" sem a tradução pq diz respeito ao conceito de Web Components (https://www.webcomponents.org/) e não de "componentes web" de uma forma mais genérica, que poderia englobar tanto os próprios componentes React quanto até mesmo as directives do Vue ou do Angular

Reviewer: @WeslleyNasRocha tu vai fazer esse ajuste aqui :point_up_2:

Author: feito
